open_city_profile/graphene.py

```diff
@@ -70,3 +77,57 @@ def resolve(self, next, root, info, **kwargs):
             self.cached_loaders = True
 
         return next(root, info, **kwargs)
+
+
+TranslationLanguageType = graphql.GraphQLEnumType(
+    name="TranslationLanguage",
+    values=OrderedDict(
+        {
+            lang[0].upper(): graphql.GraphQLEnumValue(lang[0], description=lang[1])
+            for lang in settings.LANGUAGES
+        }
+    ),
+)
+
+
+TranslationDirective = graphql.GraphQLDirective(
+    name="translation",
+    locations=[graphql.DirectiveLocation.FIELD],
+    args={
+        "in": graphql.GraphQLArgument(
+            graphql.GraphQLNonNull(TranslationLanguageType), description="The language"
+        ),
+    },
+    description="Get value translated in language",
+)
+
+
+class TranslationMiddleware:
+    def resolve(self, next, root, info, **kwargs):
+        directives = info.field_asts[0].directives
+
+        if not directives:
+            return next(root, info, **kwargs)
```

Comment:
Seems like `directives` is always an array, even if it's empty. In that case **this** early return wouldn't be strictly needed. The execution could continue to the next step and when no `translation_directives` are found, **then** return early.

Reply:
True. I like to return early if nothing to be done. But I can remove it.